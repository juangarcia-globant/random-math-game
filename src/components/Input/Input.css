
.input {
  padding: 1em;
  border-radius: 6px;
  border: 2px solid #002d87;
  font-size: 1em;
  font-weight: bold;
  text-align: center;
}

.input-sm {
  padding: 0.3rem;
}

.input-error {
  color: red;
  font-size: 0.6em;
}