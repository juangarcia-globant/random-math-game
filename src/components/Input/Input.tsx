import './Input.css';

type InputProps = {
  type?: string,
  placeholder?: string,
  size?: string,
  error?: string,
  hasError?: boolean,
  onKeyDown?: (evt: React.InputHTMLAttributes<HTMLInputElement>) => void,
  onInputChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void,
};

export const Input: React.FunctionComponent<InputProps> = ({
  type,
  placeholder,
  size = 'sm',
  error,
  hasError,
  onKeyDown,
  onInputChange,
}) => {
  return (
    <>
      <input 
        id="textId"
        type={ type || "text" } 
        className={`input input-${size}`} 
        placeholder={placeholder} 
        onChange={onInputChange}
        onKeyDown={onKeyDown}
      />
      {
        hasError && <p className='input-error'> {error} </p>
      }
    </>
  )
}

export default Input;
