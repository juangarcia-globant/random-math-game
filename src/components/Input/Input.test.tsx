import { render, screen, fireEvent } from '@testing-library/react';
import Input from './Input';

describe("<Input />", () => {
  test("Matches the snapshot", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Input  should be render", () => {
    render(<Input placeholder='placeholder' error='Error message' hasError={true} />);
    expect(screen.getByPlaceholderText('placeholder')).toBeInTheDocument();
  });

  test("Error message should be render", () => {
    render(<Input placeholder='placeholder' error='Error message' hasError={true} />);
    expect(screen.getByText('Error message')).toBeInTheDocument();
  });
});