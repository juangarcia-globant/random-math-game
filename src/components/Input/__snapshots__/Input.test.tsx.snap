// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Matches the snapshot 1`] = `
<DocumentFragment>
  <input
    class="input input-sm"
    id="textId"
    type="text"
  />
</DocumentFragment>
`;
